packages/hdwallet-keepkey/src/binance.ts

```diff
@@ -45,16 +45,22 @@ export async function binanceSignTx(
     );
     if (resp.message_type === Core.Events.FAILURE) throw resp;
 
+    const intRegex = /^(0|([1-9][0-9]*))$/;
+    const outputAmount = message.outputs[0].coins[0].amount.toString();
+    if (!intRegex.test(outputAmount)) throw new Error("output amount should be an integer");
+    const inputAmount = message.inputs[0].coins[0].amount.toString();
+    if (!intRegex.test(inputAmount)) throw new Error("input amount should be an integer");
+
     let coinOut = new BinanceTransferMsg.BinanceCoin();
-    coinOut.setAmount(message.outputs[0].coins[0].amount.toString());
+    coinOut.setAmount(outputAmount);
     coinOut.setDenom(message.outputs[0].coins[0].denom);
 
     let outputs = new BinanceTransferMsg.BinanceInputOutput();
     outputs.setAddress(message.outputs[0].address);
     outputs.setCoinsList([coinOut]);
 
     let coinIn = new BinanceTransferMsg.BinanceCoin();
-    coinIn.setAmount(message.inputs[0].coins[0].amount.toString());
+    coinIn.setAmount(inputAmount);
```

Comment:
If using my previous suggestion
```suggestion
    coinIn.setAmount(inputAmount.toString());
```